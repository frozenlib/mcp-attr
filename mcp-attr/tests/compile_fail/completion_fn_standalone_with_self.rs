use mcp_attr_macros::complete_fn;

struct TestServer {
    name: String,
}

impl TestServer {
    #[complete_fn]
    async fn complete_with_self(&self, value: &str) -> Result<Vec<String>, Box<dyn std::error::Error + Send + Sync>> {
        Ok(vec![format!("completion for {}", value)])
    }
}

fn main() {}