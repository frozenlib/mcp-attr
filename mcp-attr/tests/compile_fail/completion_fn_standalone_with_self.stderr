error: completion functions using #[complete_fn] attribute cannot have `self` parameter. Use the function inside #[mcp_server] impl block instead.
 --> tests/compile_fail/completion_fn_standalone_with_self.rs:9:34
  |
9 |     async fn complete_with_self(&self, value: &str) -> Result<Vec<String>, Box<dyn std::error::Error + Send + Sync>> {
  |                                  ^^^^
